rfc/src/rfcs/0003-cover-statement.md

```diff
@@ -0,0 +1,105 @@
+- **Feature Name:** Cover statement `cover_statement`
+- **Feature Request Issue:** <https://github.com/model-checking/kani/issues/696>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/1906>
+- **Status:** Under Review
+- **Version:** 0
+- **Proof-of-concept:** *Optional field. If you have implemented a proof of concept, add a link here*
+
+## Summary
+
+A new Kani API that allows users to check that a certain condition can occur at a specific location in the code.
+
+## User Impact
+
+Users typically want to gain confidence that a proof checks what it is supposed to check, i.e. that properties are not passing vacuously due to an over-constrained environment.
+
+A new Kani macro, `cover` will be created that can be used for checking that a certain condition _can_ occur at a specific location in the code.
```

Comment:
Something that isn't clear to me after reading this section is how `cover` is different from `assert`. If my understanding is correct, `assert` checks a condition, while `cover` checks a condition AND ensures that the statement itself is reachable.

If we agree on that, then the "_can_" in this sentence should be "_will_" instead.

Reply:
I think conceptually they are similar. `assert` says this condition must hold for _all_ executions. `cover` says this condition must hold for _at least one_ execution.